### PR TITLE
attempt at fixing travis-ci build and test checks

### PR DESCRIPTION
**Pull request by basisbit**
### What does this PR do?
Hopefully fixes the travis-ci autobuild/-test issues.

### Related Issue(s)

this is part of #1 

### Motivation

Automatic building and reporting simplifies developers' and project maintainers' life by reducing workload.

### More

- [ ] please do not merge yet, this is still work in progress

### CODE REVIEW

**Comment by basisbit** [comment]
Travis-CI is finally green 🥂

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-language: generic
-
-# use a Mac OSX image
-os: osx
-
+sudo: required
+language: bash
+env:
+  global:
+  - IMAGE_NAME=gableroux/unity3d:2018.2.6f1
```
Comment (barbeque-squared) [inline .travis.yml]: Contrary to what https://github.com/UltraStar-Deluxe/Play/wiki/Compiling-the-game and #28 say, this uses a more recent Unity version. Wiki edits are probably separate from the codebase, and I'm guessing the newer version is deliberate here, but don't forget to update the PR and wiki after merging.

File: UltraStar Play/Assets/Editor/BuildCommand.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
+﻿using UnityEditor;
```
Comment (barbeque-squared) [inline UltraStar Play/Assets/Editor/BuildCommand.cs]: This file has mixed tabs and spaces (I'm surprised that Codacy didn't pick up on it?). Should be space-only.

File: tools/travis/before_script.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -e
+set -x
+mkdir -p /root/.cache/unity3d
+mkdir -p /root/.local/share/unity3d/Unity/
```
Comment (barbeque-squared) [inline tools/travis/before_script.sh]: One mkdir ends without a trailing slash, the other with. I know that both do exactly the same, though I usually try to keep them consistent within the same file, especially if they're two consecutive lines.

File: tools/travis/build.sh [hunk 4]
```diff
@@ -1,57 +1,36 @@
...
+  echo "Run succeeded, no failures occurred";
+elif [ $UNITY_EXIT_CODE -eq 2 ]; then
+  echo "Run succeeded, some tests failed";
+elif [ $UNITY_EXIT_CODE -eq 3 ]; then
+  echo "Run failure (other failure)";
+else
+  echo "Unexpected exit code $UNITY_EXIT_CODE";
+fi
+
+ls -la "UltraStar Play/$BUILD_PATH"
```
Comment (barbeque-squared) [inline tools/travis/build.sh]: Is the output of this ls used somewhere? In understand why the echo's are here, just not what this ls is doing here.

File: tools/travis/activat.ion [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            <Feature Value="19"/>
+            <Feature Value="62"/>
+        </Features>
+        <DeveloperData Value="AQAAAEY0LVI4VEstNU1SOS1UVUNVLUtXOFgtWUM1Mg=="/>
+        <SerialMasked Value="F4-R8TK-5MR9-TUCU-KW8X-XXXX"/>
+        <StartDate Value="2017-12-16T14:55:19"/>
+        <UpdateDate Value="2018-11-02T23:05:29"/>
+        <InitialActivationDate Value="2017-12-16T14:55:19"/>
+        <LicenseVersion Value="6.x"/>
+        <ClientProvidedVersion Value="2018.2.6f1"/>
```
Comment (barbeque-squared) [inline tools/travis/activat.ion]: See the other comment about the Unity version.

File: tools/travis/build.sh (hunk 4 above)
Comment (basisbit) [reply tools/travis/build.sh]: the folder output is only there to simplify analysis if travis-ci build didn't work as intended or if if says "green" but actually should report "red"

File: tools/travis/activat.ion (hunk 5 above)
Comment (basisbit) [reply tools/travis/activat.ion]: Wasn't able to get the older versions to work with travis-ci. It supports the desired testing + compile checks for CI purposes of the 2018.1.4f1 version project files. Also, see #28

File: UltraStar Play/Assets/Editor/BuildCommand.cs (hunk 2 above)
Comment (basisbit) [reply UltraStar Play/Assets/Editor/BuildCommand.cs]: sorry, my fault. Neither the compiler nor most modern IDEs care about tabs or spaces at line beginning, so I didn't notice the difference. I'll update the PR.

Comment (barbeque-squared) [reply UltraStar Play/Assets/Editor/BuildCommand.cs]: Looks good now.

File: tools/travis/build.sh (hunk 4 above)
Comment (barbeque-squared) [reply tools/travis/build.sh]: Had a feeling it was something like that (but had no way to be sure), so no problem leaving it in.

File: tools/travis/activat.ion (hunk 5 above)
Comment (barbeque-squared) [reply tools/travis/activat.ion]: Green light for using it then. I use 2018.2.0f2 myself, and as long as I don't commit anything involving .meta files, I haven't noticed any issues on the current .NET.

**Pull request merged**